java/arcs/core/storage/referencemode/Message.kt

```diff
@@ -112,30 +109,6 @@ sealed class Message(
 }
 
 /** Converts a general [ProxyMessage] into a reference mode-safe [ProxyMessage]. */
-fun ProxyMessage<CrdtData, CrdtOperation, Any?>.toReferenceModeMessage():
-    ProxyMessage<CrdtData, CrdtOperationAtTime, Referencable> {
-    return when (this) {
-        is ProxyMessage.ModelUpdate ->
-            ProxyMessage.ModelUpdate(model, id)
-        is ProxyMessage.Operations ->
-            ProxyMessage.Operations(operations.toReferenceModeMessageOps(), id)
-        is ProxyMessage.SyncRequest -> ProxyMessage.SyncRequest(id)
-    }
-}
+fun ProxyMessage<CrdtData, CrdtOperation, Any?>.toReferenceModeMessage()
+    =  this as ProxyMessage<CrdtData, CrdtOperationAtTime, Referencable>
 
-@Suppress("UNCHECKED_CAST")
-private fun List<CrdtOperation>.toReferenceModeMessageOps(): List<CrdtOperationAtTime> {
-    return this.map { op ->
-        when (op) {
-            is CrdtSingleton.Operation.Update<*> ->
-                CrdtSingleton.Operation.Update(op.actor, op.clock, op.value)
-            is CrdtSingleton.Operation.Clear<*> ->
-                CrdtSingleton.Operation.Clear<Referencable>(op.actor, op.clock)
-            is CrdtSet.Operation.Add<*> ->
-                CrdtSet.Operation.Add(op.actor, op.clock, op.added)
-            is CrdtSet.Operation.Remove<*> ->
-                CrdtSet.Operation.Add(op.actor, op.clock, op.removed)
```

Comment:
This translation error also caused an issue.

These two functions were making copies of ProxyMessages in order to cast to the correct type. But due to type erasure, the copying is not needed; the message can simply be cast to the correct type.

Reply:
Actually we could even remove this message if we change CrdtOperation -> CrdtOperationAtTime everywhere, but I didn't want to let this change balloon too much.